packages/ui-i18n/src/bidirectional.js

```diff
@@ -69,22 +71,41 @@ const bidirectional = decorator((ComposedComponent) => {
       ...TextDirectionContext.types
     }
 
-    get dir () {
+    get dir() {
       const context = getTextDirectionContext(this.context) || {}
-      return context.dir  || this.props.dir || getTextDirection()
+      return context.dir || this.props.dir || getTextDirection()
     }
 
-    get rtl () {
+    get rtl() {
       return this.dir === DIRECTION.rtl
     }
 
-    get ltr () {
+    get ltr() {
       return this.dir === DIRECTION.ltr
     }
   }
 })
 
 bidirectional.DIRECTION = DIRECTION
 
+const useBidirectional = (props = {}) => {
+  const context = useTextDirectionContext()
+  const dir = props.dir || context
+
+  return {
+    dir,
+    rtl: dir === DIRECTION.rtl,
+    ltr: dir === DIRECTION.ltr
+  }
+}
+
+const withBidirectional = (ComposedComponent) => {
+  return function BidirectionalComponent(props) {
```

Comment:
What is the purpose in naming the function and using the "function" prefix?  Why not simply return an anonymous arrow function?

Reply:
Since this function will be used as a React component and we have an ESLint rule which doesn't allow nameless components I had to give it some sort of name.
`Component definition is missing display name eslintreact/display-name`